rskj-core/src/main/java/org/ethereum/rpc/parameters/BlockRefParam.java

```diff
@@ -0,0 +1,85 @@
+package org.ethereum.rpc.parameters;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import org.ethereum.rpc.BlockRef;
+import org.ethereum.rpc.exception.RskJsonRpcRequestException;
+import org.ethereum.util.Utils;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@JsonDeserialize(using = BlockRefParam.Deserializer.class)
+public class BlockRefParam implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private static final List<String> IDENTIFIERS_TO_VALIDATE = Arrays.asList("earliest", "latest", "pending");
+    private static final List<String> BLOCK_INPUT_KEYS_TO_VALIDATE = Arrays.asList("blockHash", "blockNumber");
+
+    private final String identifier;
+    private final Map<String, String> inputs;
+
+    public BlockRefParam(BlockRef blockRef) {
+        if(blockRef.getIdentifier() != null) {
+            validateString(blockRef.getIdentifier());
+        } else {
+            validateMap(blockRef.getInputs());
+        }
+
+        this.identifier = blockRef.getIdentifier();
+        this.inputs = blockRef.getInputs();
+    }
+
+    private void validateString(String identifier) {
+        if(!IDENTIFIERS_TO_VALIDATE.contains(identifier)
+                && !Utils.isDecimalString(identifier)
+                && !Utils.isHexadecimalString(identifier)) {
+            throw RskJsonRpcRequestException.invalidParamError("Invalid block identifier '" + identifier + "'");
+        }
+    }
+
+    private void validateMap(Map<String, String> inputs) {
```

Comment:
As commented above this implementation does not support `requireCanonical` functionality